### PR TITLE
HOMING: Allow grbl to pend on a homing request

### PR DESCRIPTION
**Pull request by Jeff-Ciesielski**

### CODE REVIEW

**Comment by Jeff-Ciesielski** [comment]
@StephanCoetzee 

This allows us to put homing requests 'in the pipe' with normal gcode commands.  The homing command will wait for all previous commands before executing.

File: protocol.c [hunk 1]
```diff
@@ -42,10 +42,10 @@ static uint8_t next_report=REQUEST_STATUS_REPORT;
 // Directs and executes one line of formatted input from protocol_process. While mostly
 // incoming streaming g-code blocks, this also directs and executes Grbl internal commands,
 // such as settings, initiating the homing cycle, and toggling switch states.
-static void protocol_execute_line(char *line)
+static uint8_t protocol_execute_line(char *line)
 {
   protocol_execute_runtime(); // Runtime command check point.
-  if (sys.abort) { return; } // Bail to calling function upon system abort
+  if (sys.abort) { return STATUS_ABORT; } // Bail to calling function upon system abort
```
Comment (StephanCoetzee) [inline protocol.c]: Put the return on the next line. 
I try to format all the code that we touch correctly.

File: system.c [hunk 2]
```diff
@@ -198,6 +198,40 @@ uint8_t system_execute_line(char *line)
         report_feedback_message(MESSAGE_ENABLED);
       }
       break;
+        case 'H' : // Perform homing cycle [IDLE/ALARM], only if idle or lost
+	  if ( !(sys.state == STATE_IDLE || sys.state == STATE_ALARM) ) {
```
Comment (StephanCoetzee) [inline system.c]: Extra spaces after ( and before )

Unless you're into that kind of thing

File: system.c [hunk 3]
```diff
@@ -198,6 +198,40 @@ uint8_t system_execute_line(char *line)
         report_feedback_message(MESSAGE_ENABLED);
       }
       break;
+        case 'H' : // Perform homing cycle [IDLE/ALARM], only if idle or lost
```
Comment (StephanCoetzee) [inline system.c]: Is this indentation correct?

File: protocol.c [hunk 4]
```diff
@@ -111,7 +112,9 @@ void protocol_main_loop()
     while(progman_read(&c)) {
       if ((c == '\n') || (c == '\r')) { // End of line reached
         line[char_counter] = 0; // Set string termination character.
-        protocol_execute_line(line); // Line is complete. Execute it!
+        while(protocol_execute_line(line) == STATUS_IDLE_WAIT) {
```
Comment (StephanCoetzee) [inline protocol.c]: Maybe add a comment to explain a bit.

**Comment by Jeff-Ciesielski** [comment]
Updated

**Comment by StephanCoetzee** [comment]
LGTM

**Review by StephanCoetzee — APPROVED** [review]
LGTM

**Pull request merged**